src/main/java/yevhen/bookstore/service/impl/BookServiceImpl.java

```diff
@@ -0,0 +1,51 @@
+package yevhen.bookstore.service.impl;
+
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import yevhen.bookstore.dto.BookDto;
+import yevhen.bookstore.dto.CreateBookRequestDto;
+import yevhen.bookstore.mapper.BookMapper;
+import yevhen.bookstore.model.Book;
+import yevhen.bookstore.repository.BookRepository;
+import yevhen.bookstore.service.BookService;
+
+@RequiredArgsConstructor
+@Service
+public class BookServiceImpl implements BookService {
+
+    private final BookRepository bookRepository;
+    private final BookMapper bookMapper;
+
+    @Override
+    public BookDto save(CreateBookRequestDto requestDto) {
+        Book book = bookMapper.toModel(requestDto);
+        return bookMapper.toDto(bookRepository.save(book));
+    }
+
+    @Override
+    public List<BookDto> findAll() {
+        return bookRepository.findAll()
+                .stream()
+                .map(bookMapper::toDto)
+                .toList();
+    }
+
+    @Override
+    public BookDto getBookById(Long id) {
+        Book book = bookRepository.getBookById(id);
+        return bookMapper.toDto(book);
+    }
```

Comment:
You need use in this place findById method from repository with Optional<Book> return type. Because you don't handle cases when book by special id doesn't exist in DB